### ISSUE TITLE
undefined method `join' for #<String... - On web-interface When redis-server not running

### ISSUE THREAD

**Issue by IAmFledge** [resque/resque#1286]
This is primarily a call for a more graceful fail message, and also for anybody Googling this issue.

If redis-server is not running, hitting the resque-web page will fail with: `undefined method `join' for #<String...`

The error is actually ocurring in Sinatra

`Sinatra::ShowExceptions#call
sinatra (1.4.5) lib/sinatra/show_exceptions.rb, line 37`

### If you're encountering this error, ensure that the redis server is running.

Version info:
```
resque (1.25.2)
      mono_logger (~> 1.0)
      multi_json (~> 1.0)
      redis-namespace (~> 1.3)
      sinatra (>= 0.9.2)
      vegas (~> 0.1.2)
resque-scheduler (4.0.0)
      mono_logger (~> 1.0)
      redis (~> 3.0)
      resque (~> 1.25)
      rufus-scheduler (~> 3.0)*
```

**Comment by steveklabnik** [comment]
Any idea on how to reproduce the error in the first place?

**Comment by dfockler** [comment]
I reproduced this by changing my Sinatra version from 1.4.6 to 1.4.5. It doesn't happen with the latest version (1.4.6) of Sinatra. It gives a better Redis::CannotConnectError instead.

**Comment by steveklabnik** [comment]
So, I guess the question is, do we want to pursue a fix here, or just say that older versions may have bad error messages?

**Comment by steveklabnik** [comment]
We could also bump the minimum required sinatra version, I guess?

**Comment by dfockler** [comment]
FWIW It's a known issue in Sinatra that they fixed when they bumped up to 1.4.6
https://github.com/sinatra/sinatra/issues/986

**Comment by steveklabnik** [comment]
Let's give this a close, then, as it's really their issue.
